insights/tests/client/connection/test_checkin.py

```diff
@@ -191,3 +191,14 @@ def test_response_failure(get_proxies, post, get_canonical_facts, rm_conf, statu
 
     with raises(Exception):
         connection.checkin()
+
+
+def test_unregistered_host():
+    """
+    If the host is not registered, the checkin fails.
+    """
+    config = InsightsConfig(legacy_upload=False, base_url="www.example.com")
+    connection = InsightsConnection(config)
```

Comment:
Hmm this test is not mocking things like all the other tests in this file; this probably means that it will actually try to get the facts, read the config, and get the machine-id of the system, thus potentially failing the test if run on a system registered with insights-client.